### PR TITLE
chore: import aioeapi module

### PR DESCRIPTION
**Pull request by mtache**
Import aioeapi module from [Jeremy Schulman](https://github.com/jeremyschulman) repository at https://github.com/jeremyschulman/aio-eapi

Update the codebase to comply with ANTA's linting rules.

Also populate the eAPI id field with "ANTA-<command ID>" or "ANTA-<collection ID>-<command ID>".
When running NRFU, the collection ID is the test name.

This allows for easier eAPI requests debug on the device using EOS configuration: `trace CapiApp setting UwsgiRequestContext/4,CapiUwsgiServer/4`

```
bash tail -f /var/log/agents/CapiApp-*

2024-05-15 15:32:54.056166  1429 UwsgiRequestContext  4 request content b'{"jsonrpc": "2.0", "method": "runCmds", "params": {"version": "latest", "cmds": [{"cmd": "show ip route vrf default 10.255.0.3", "revision": 4}], "format": "json", "autoComplete": false, "expandAliases": false}, "id": "ANTA-VerifyRoutingTableEntry-132366530677328"}'
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: asynceapi/config_session.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+#                                 CODE BEGINS
+#
+# -----------------------------------------------------------------------------
+
+
+class SessionConfig:
+    """
+    Send configuration to a device using the EOS session mechanism.
+
+    This is the preferred way of managing configuraiton changes.
```
Comment (carl-baillargeon) [inline asynceapi/config_session.py:31]:
```suggestion
    This is the preferred way of managing configuration changes.
```

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -77,3 +77,14 @@ Example:
 ```bash
 ANTA_DEBUG=true anta -l DEBUG --log-file anta.log nrfu --enable --username username --password arista --inventory inventory.yml -c nrfu.yml text
 ```
+
+### Troubleshooting on EOS
+
+ANTA is using a specific ID in eAPI requests towards EOS. This allows for easier eAPI requests debug on the device using EOS configuration `trace CapiApp setting UwsgiRequestContext/4,CapiUwsgiServer/4` to set up CapiApp agent logs.
```
Comment (gmuloc) [inline docs/troubleshooting.md:83]:
```suggestion
ANTA is using a specific ID in eAPI requests towards EOS. This allows for easier eAPI requests debugging on the device using EOS configuration `trace CapiApp setting UwsgiRequestContext/4,CapiUwsgiServer/4` to set up CapiApp agent logs.
```

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -77,3 +77,14 @@ Example:
...
+
+Then, you can view agent logs using:
+```
```
Comment (gmuloc) [inline docs/troubleshooting.md:86]:
```suggestion
```bash
```

File: anta/device.py [hunk 4]
```diff
@@ -131,11 +132,11 @@ async def _collect(self, command: AntaCommand) -> None:
 
         Args:
         ----
-            command: the command to collect
-
+            command: The command to collect.
+            collection_id: An identifier that will used to build the eAPI request ID.
```
Comment (gmuloc) [inline anta/device.py:136]:
```suggestion
            collection_id: An identifier that is used to build the eAPI request ID.
```

File: anta/device.py [hunk 5]
```diff
@@ -148,8 +149,8 @@ async def collect(self, command: AntaCommand) -> None:
 
         Args:
         ----
-            command (AntaCommand): The command to process.
-
+            command: The command to collect.
+            collection_id: An identifier that will used to build the eAPI request ID.
```
Comment (gmuloc) [inline anta/device.py:153]:
```suggestion
            collection_id: An identifier that is used to build the eAPI request ID.
```

File: anta/device.py [hunk 6]
```diff
@@ -161,20 +162,20 @@ async def collect(self, command: AntaCommand) -> None:
...
-    async def collect_commands(self, commands: list[AntaCommand]) -> None:
+    async def collect_commands(self, commands: list[AntaCommand], *, collection_id: str | None = None) -> None:
         """Collect multiple commands.
 
         Args:
         ----
-            commands: the commands to collect
-
+            commands: The commands to collect.
+            collection_id: An identifier that will used to build the eAPI request ID.
```
Comment (gmuloc) [inline anta/device.py:176]:
```suggestion
            collection_id: An identifier that is used to build the eAPI request ID.
```

File: anta/device.py [hunk 7]
```diff
@@ -314,9 +315,10 @@ async def _collect(self, command: AntaCommand) -> None:  # noqa: C901  function
 
         Args:
         ----
-            command: the AntaCommand to collect.
+            command: The command to collect.
+            collection_id: An identifier that will used to build the eAPI request ID.
```
Comment (gmuloc) [inline anta/device.py:319]:
```suggestion
            collection_id: An identifier that is used to build the eAPI request ID.
```

File: asynceapi/__init__.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Initially written by Jeremy Schulman at https://github.com/jeremyschulman/aio-eapi
```
Comment (gmuloc) [inline asynceapi/__init__.py:1]: Apache2 needs a double copyright one for Jeremy Schulman and ours in these files

**Review by gmuloc — COMMENTED** [review]
Let s add a kudo to Jerermy and a link in our README

**Review by gmuloc — APPROVED** [review]
🚢 :shipit:

**Pull request merged**